sktime/clustering/_time_series_k_means.py

```diff
@@ -0,0 +1,159 @@
+# -*- coding: utf-8 -*-
+"""Time Series K-Means Clusterer."""
+
+__author__ = ["Christopher Holder", "Tony Bagnall"]
+__all__ = ["TimeSeriesKMeans"]
+
+from typing import List
+from sktime.clustering.base.base_types import (
+    Metric_Parameter,
+    Data_Frame,
+    Numpy_Array,
+)
+from sktime.clustering.base.base import (
+    BaseCluster,
+    Init_Algo,
+    Averaging_Algo,
+    Averaging_Algo_Dict,
+)
+from sktime.clustering.partitioning._averaging_metrics import (
+    BarycenterAveraging,
+    MeanAveraging,
+)
+from sktime.clustering.partitioning._time_series_k_partition import TimeSeriesKPartition
+
+
+class TimeSeriesKMeans(TimeSeriesKPartition, BaseCluster):
+    """Time Series K-Means Clusterer.
+
+    This is a work in progress.
+    """
+
+    __averaging_algorithm_dict: Averaging_Algo_Dict = {
+        "dba": BarycenterAveraging,
+        "mean": MeanAveraging,
+    }
+
+    def __init__(
+        self,
+        n_clusters: int = 8,
+        init_algorithm: Init_Algo = "random",
```

Comment:
For types and classes can we just use camel case here and throughout the code? 
```suggestion
        init_algorithm: InitAlgo = "random",
```